presto-session-property-managers/src/main/java/io/prestosql/plugin/session/AbstractSessionPropertyManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.session;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.spi.session.SessionConfigurationContext;
+import io.prestosql.spi.session.SessionPropertyConfigurationManager;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public abstract class AbstractSessionPropertyManager
+        implements SessionPropertyConfigurationManager
```

Comment:
Please extract `AbstractSessionPropertyManager` extraction as separate commit. So logical changes are separated from refactors. Now it is more difficult to track how it changed.

Reply:
Cut the change in 4 commits to ease the review

Reply:
Thanks!